src/components/Icon/IconStyles.tsx

```diff
@@ -5,28 +5,34 @@ import { Spacing } from '../../types/Spacing'
 import { marginCss } from '../common/Spacing/SpacingStyles'
 
 interface StyledIconProps {
-	size: Spacing | string | number
+	$size: Spacing | string | number
 }
 
-export const StyledIcon = styled(SVG)<StyledIconProps>`
-	display: inline-block;
-	${({ size, theme }): string => {
-		if (Spacing.includes(size as unknown as Spacing)) {
+export const StyledIconContainer = styled.span<StyledIconProps>`
+	display: inline-flex;
+
+	${({ $size, theme }): string => {
+		if (Spacing.includes($size as unknown as Spacing)) {
```

Comment:
Would it be cleaner to create a type guard for this?

```
const isSpacing = (size: Spacing | string | number): size is Spacing => {
  return Spacing.includes($size as unknown as Spacing)
}
```

And then using the type guard in a if statement should always infer the correct type

```
if (isSpacing($size)) {
// $size should be Spacing
} else if (typeof $size === 'number') {
// $size should be number
}
```

Reply:
Wow, TBH, I had no idea about the concept of "type guards" 😀  Pretty cool. And actually this piece of code was used in two places so this function was exactly what it needed. Thanks.